src/sdam/server_description.ts

```diff
@@ -53,6 +53,7 @@ export class ServerDescription {
   passives: string[];
   arbiters: string[];
   tags: TagSet;
+  // TODO: this error type is not actually guaranteed
   error: MongoServerError | null;
```

Comment:
it might not be guaranteed depending on what connection checkout returns, which relies on the error returned from `connect`, which is typed as `AnyError`, so this should be typed as `AnyError`, too